From 9ded92650f9de4d8e03faa370972f29d7fee0f05 Mon Sep 17 00:00:00 2001
From: Yanmin Zhang <yanmin.zhang@intel.com>
Date: Thu, 30 Aug 2012 03:49:51 +0800
Subject: [PORT FROM MAIN] wlan: Fix div by 0

BZ: 54567

Fix it by checking wl->beacon_int==0. If it is, change it to
default value.

Change-Id: I648900b1aeaad6fc70e4efe4b7377fac617c5bee
Orig-Change-Id: I306ef086a2e728d5550d9b2d92785c8740877860
Signed-off-by: Yanmin Zhang <yanmin.zhang@intel.com>
Signed-off-by: Aymen Zayet <aymen.zayet@intel.com>
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Reviewed-on: http://android.intel.com:8080/66517
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/77988
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/main.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
index 5ba38fd..3a58f35 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
@@ -4286,6 +4286,10 @@ static int wl1271_bss_beacon_info_changed(struct wl1271 *wl,
 			bss_conf->beacon_int);
 
 		wlvif->beacon_int = bss_conf->beacon_int;
+		if (!wlvif->beacon_int) {
+			WARN(1, "wl->beacon_int is 0\n");
+			wlvif->beacon_int = WL1271_DEFAULT_BEACON_INT;
+		}
 	}
 
 	if ((changed & BSS_CHANGED_AP_PROBE_RESP) && is_ap) {
@@ -4577,6 +4581,10 @@ sta_not_found:
 			int ieoffset;
 			wlvif->aid = bss_conf->aid;
 			wlvif->beacon_int = bss_conf->beacon_int;
+			if (!wlvif->beacon_int) {
+				WARN(1, "wl->beacon_int is 0\n");
+				wlvif->beacon_int = WL1271_DEFAULT_BEACON_INT;
+			}
 			set_assoc = true;
 
 			/*
-- 
1.7.4.1

